### PR TITLE
renderer: add text trim with option to disable

### PR DESCRIPTION
**Pull request by davc0n**
ref. https://github.com/hyprwm/hyprlock/issues/284

Not sure if that's the best way to trim a string in C++, please let me know if there's a better solution.

If it will be accepted I'll update the wiki too.

### CODE REVIEW

File: src/config/ConfigManager.cpp [hunk 1]
```diff
@@ -44,6 +44,7 @@ void CConfigManager::init() {
     m_config.addSpecialConfigValue(name, "shadow_boost", Hyprlang::FLOAT{1.2});
 
     m_config.addConfigValue("general:disable_loading_bar", Hyprlang::INT{0});
+    m_config.addConfigValue("general:disable_text_trim", Hyprlang::INT{0});
```
Comment (vaxerski) [inline src/config/ConfigManager.cpp:47]: can you flip the option? (text_trim and default 1)

**Review by vaxerski — COMMENTED** [review]
also needs a clang-format

File: src/config/ConfigManager.cpp (hunk 1 above)
Comment (davc0n) [reply src/config/ConfigManager.cpp:47]: the idea was to be coherent with other general options that are 0 by default, are you sure I should flip it?

Comment (vaxerski) [reply src/config/ConfigManager.cpp:47]: yea, people complain about having both `xxx = bool` and `disable_yyy = bool` options and call it confusing. People generally prefer `enable_xxx` options it seems.

Comment (davc0n) [reply src/config/ConfigManager.cpp]: done

**Review by vaxerski — APPROVED** [review]
lgtm thanks!

**Pull request merged**